### PR TITLE
Feat/end user/registration

### LINKED ISSUE #31: 4.2 End User Registration (6h)

### PR DESCRIPTION
**Pull request by VanHoanTran**
close #31

### CODE REVIEW

File: packages/end-user-fe/src/components/EndUserLogin/index.js [hunk 1]
```diff
@@ -1,7 +1,102 @@
...
+              variant="body2"
+            >
+              {'Forgot password?'}
+            </Link>
+            <Link
+              onClick={() => {
+                navigate(paths.registration());
+              }}
+              sx={{
+                '&:hover': { cursor: 'pointer' },
```
Comment (ghost) [inline packages/end-user-fe/src/components/EndUserLogin/index.js:89]: No need for the '&:hover', see comments on the other PR

File: packages/end-user-fe/src/paths.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ const login = () => ({ pathname: '/login' });
 
 const registration = () => ({ pathname: '/registration' });
 
+const profile = ({ id = ':id' } = {}) => ({
```
Comment (ghost) [inline packages/end-user-fe/src/paths.js:5]: No need for the id, we will only ever show the current user in the "end-user" interface

File: packages/end-user-fe/src/resources/endUserApi.js [hunk 3]
```diff
@@ -26,3 +26,80 @@ export const fetchCurrentUser = apiCall(() => ({
 export const fetchShopSettings = apiCall(() => ({
   url: '/shop',
 }));
+// Users
+export const fetchUserDetails = apiCall(({ id }) => ({
```
Comment (ghost) [inline packages/end-user-fe/src/resources/endUserApi.js:30]: No need for ID, the BE will know which user to serve based on the token

File: packages/end-user-fe/src/resources/endUserApi.js [hunk 4]
```diff
@@ -26,3 +26,80 @@ export const fetchCurrentUser = apiCall(() => ({
...
+    password,
+    email,
+    phone,
+    address1,
+    address2,
+    zip,
+    city,
+    state,
+    country,
+    ...rest
```
Comment (ghost) [inline packages/end-user-fe/src/resources/endUserApi.js:49]: I don't think you are using this

File: packages/end-user-fe/src/resources/endUserApi.js [hunk 5]
```diff
@@ -26,3 +26,80 @@ export const fetchCurrentUser = apiCall(() => ({
...
```
Comment (ghost) [inline packages/end-user-fe/src/resources/endUserApi.js:84]: I don't think you are using this

File: packages/end-user-fe/src/resources/endUserApi.js [hunk 6]
```diff
@@ -26,3 +26,80 @@ export const fetchCurrentUser = apiCall(() => ({
...
+        state,
+        country,
+      },
+    },
+  })
+);
+
+export const updateUser = apiCall(
+  ({
+    id,
```
Comment (ghost) [inline packages/end-user-fe/src/resources/endUserApi.js:73]: No need for the ID, BE will know what user are we talking about based on the token

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/end-user-fe/src/resources/endUserApi.js (hunk 4 above)
Comment (VanHoanTran) [reply packages/end-user-fe/src/resources/endUserApi.js:49]: I was thinking about these attributes in the schema `isVerified: Boolean` and `verificationCode: String`. But, actually, I did not use it. I will drop those.

File: packages/end-user-fe/src/resources/endUserApi.js (hunk 3 above)
Comment (VanHoanTran) [reply packages/end-user-fe/src/resources/endUserApi.js:30]: so we are using the same API for the current user and the fetchUserDetails. am I correct?

Comment (ghost) [reply packages/end-user-fe/src/resources/endUserApi.js:30]: Yes, in the end user FE the user in question is always the current user

**Pull request merged**